src/Dependency/InheritanceFlatter.php

```diff
@@ -13,13 +13,16 @@ public function flattenDependencies(
         Result $dependencyResult
     ): void {
         foreach ($astMap->getAstClassReferences() as $classReference) {
-            foreach ($astMap->getClassInherits($classReference->getClassLikeName()) as $inherit) {
+            $tokenLikeName = $classReference->getTokenLikeName();
+            foreach ($astMap->getClassInherits($tokenLikeName) as $inherit) {
                 foreach ($dependencyResult->getDependenciesByClass($inherit->getClassLikeName()) as $dep) {
+                    $depTokenName = $dep->getTokenLikeNameB();
+                    assert($depTokenName instanceof AstMap\ClassLikeName);
```

Comment:
Why do we need this assert?

Reply:
This is not done yet, it will probably look different, right now it ti there for consistency/passing Static analysis.